crates/redeem/src/lib.rs

```diff
@@ -568,15 +589,32 @@ impl<T: Config> Module<T> {
 
         ensure!(redeem.vault == vault_id, Error::<T>::UnauthorizedUser);
 
-        ext::vault_registry::try_increase_to_be_issued_tokens::<T>(&vault_id, redeem.amount_btc)?;
-        ext::vault_registry::issue_tokens::<T>(&vault_id, redeem.amount_btc)?;
-        ext::treasury::mint::<T>(vault_id, redeem.amount_btc);
+        let reimbursed_amount = redeem
+            .amount_btc
+            .checked_add(&redeem.transfer_fee_btc)
+            .ok_or(Error::<T>::ArithmeticOverflow)?;
+
+        ext::vault_registry::try_increase_to_be_issued_tokens::<T>(&vault_id, reimbursed_amount)?;
+        ext::vault_registry::issue_tokens::<T>(&vault_id, reimbursed_amount)?;
+        ext::treasury::mint::<T>(vault_id, reimbursed_amount);
 
         Self::set_redeem_status(redeem_id, RedeemRequestStatus::Completed);
 
         Ok(())
     }
 
+    /// get current inclusion fee based on the expected number of bytes in the transaction, and
+    /// the inclusion fee rate reported by the oracle
+    fn get_current_inclusion_fee() -> Result<PolkaBTC<T>, DispatchError> {
+        {
+            let size: u32 = Self::redeem_transaction_size();
+            let satoshi_per_bytes: u32 = ext::oracle::satoshi_per_bytes::<T>().fast;
+
+            let fee = (size as u64) * (satoshi_per_bytes as u64);
```

Comment:
Maybe we should use checked math here?

Reply:
But multiplying two `u32`s can not overflow a `u64`. Do we really want to use a zero-tolerance policy for runtime-panicable functions?

Reply:
I recall a conversation that we wanted to make it a policy to only ever use checked math to be on the safe.